From ca77f28fb1016714fc163fa5b08a9b69fc65d665 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 30 May 2022 14:19:49 +0800
Subject: [PATCH 066/274] LF-2650-1 dt-bindings: phy: Add bindings for i.MX93
 Synopsys DesignWare MIPI DPHY

This patch adds DT bindings for Freescale i.MX93
Synopsys DesignWare MIPI DPHY.

Cc: Sandor Yu <Sandor.yu@nxp.com>
Reviewed-by: Sandor Yu <Sandor.yu@nxp.com>
Signed-off-by: Liu Ying <victor.liu@nxp.com>
---
 .../bindings/phy/fsl,imx93-mipi-dphy.yaml     | 65 +++++++++++++++++++
 1 file changed, 65 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/phy/fsl,imx93-mipi-dphy.yaml

diff --git a/Documentation/devicetree/bindings/phy/fsl,imx93-mipi-dphy.yaml b/Documentation/devicetree/bindings/phy/fsl,imx93-mipi-dphy.yaml
new file mode 100644
index 000000000..222b4f9ef
--- /dev/null
+++ b/Documentation/devicetree/bindings/phy/fsl,imx93-mipi-dphy.yaml
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/phy/fsl,imx93-mipi-dphy.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Freescale i.MX93 Synopsys DesignWare MIPI DPHY
+
+maintainers:
+  - Liu Ying <victor.liu@nxp.com>
+
+properties:
+  compatible:
+    const: fsl,imx93-mipi-dphy
+
+  "#phy-cells":
+    const: 0
+
+  clocks:
+    items:
+      - description: PHY configuration clock
+      - description: PHY reference clock
+
+  clock-names:
+    items:
+      - const: phy_cfg
+      - const: phy_ref
+
+  power-domains:
+    maxItems: 1
+
+  assigned-clocks:
+    maxItems: 1
+
+  assigned-clock-parents:
+    maxItems: 1
+
+  assigned-clock-rates:
+    maxItems: 1
+
+required:
+  - compatible
+  - "#phy-cells"
+  - clocks
+  - clock-names
+  - power-domains
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/clock/imx93-clock.h>
+    #include <dt-bindings/power/imx93-power.h>
+    dphy: dphy {
+        compatible = "fsl,imx93-mipi-dphy";
+        clocks = <&clk IMX93_CLK_MIPI_PHY_CFG>, <&clk IMX93_CLK_24M>;
+        clock-names = "phy_cfg", "phy_ref";
+        assigned-clocks = <&clk IMX93_CLK_MIPI_PHY_CFG>;
+        assigned-clock-parents = <&clk IMX93_CLK_24M>;
+        assigned-clock-rates = <24000000>;
+        #phy-cells = <0>;
+        power-domains = <&media_blk_ctrl IMX93_MEDIABLK_PD_MIPI_DSI>;
+    };
+
+...
-- 
2.25.1

